oap-server/server-receiver-plugin/skywalking-management-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/register/provider/handler/v8/ManagementServiceHandler.java

```diff
@@ -47,13 +51,27 @@ public ManagementServiceHandler(ModuleManager moduleManager) {
                                           .getService(NamingControl.class);
     }
 
+    /**
+     * Identify the layer of instance. Such as ${@link Layer#FAAS}.
+     */
+    private Layer identifyInstanceLayer(String layer) {
+        try {
+            return Layer.nameOf(layer);
+        } catch (UnexpectedException e) {
```

Comment:
Using `exception` to control the process flow is not a good practice. `Layer#nameOf` throws an exception when an illegal name is correct, but also, you should have a new `Layer#has` method to avoid this catch.

Reply:
Done, set to GENERAL when layer is empty.

```java
    /**
     * Identify the layer of instance. Such as ${@link Layer#FAAS}.
     */
    private Layer identifyInstanceLayer(String layer) {
        if (StringUtil.isEmpty(layer)) {
            return Layer.GENERAL;
        } else {
            return Layer.nameOf(layer);
        }
    }
```

Reply:
Done, set to GENERAL when layer is empty.

```java
    /**
     * Identify the layer of instance. Such as ${@link Layer#FAAS}.
     */
    private Layer identifyInstanceLayer(String layer) {
        if (StringUtil.isEmpty(layer)) {
            return Layer.GENERAL;
        } else {
            return Layer.nameOf(layer);
        }
    }
```